src/NuGet.Clients/PackageManagement.PowerShellCmdlets/NuGetPowerShellBaseCommand.cs

```diff
@@ -275,24 +277,65 @@ protected override sealed void ProcessRecord()
         #region Cmdlets base APIs
 
         /// <summary>
-        /// Initializes source repositories for PowerShell cmdlets, based on config, source string, and/or host active source property value.
+        /// Looks through all available sources (including those disabled) by matching source name and url to get matching sources.
         /// </summary>
         /// <param name="source">The source string specified by -Source switch.</param>
-        protected void UpdateActiveSourceRepository(string source)
+        /// <returns></returns>
+        protected PackageSource GetMatchingSource(string source)
         {
             var packageSources = _sourceRepositoryProvider?.PackageSourceProvider?.LoadPackageSources();
```

Comment:
`_sourceRepositoryProvider` can't be null, we try to avoid overly defensive coding style.